src/app/screens/sign-up/sign-up.component.spec.ts

```diff
@@ -1,25 +1,115 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { of } from 'rxjs';
+import { FormModule } from 'src/app/components/form/form.module';
+import { FieldsMocks, SignUpMock, SignUpMockEmpty, SignUpMockErrorMessage, SignUpMockResponse } from 'src/app/helpers/mocks/sign-up.mock';
+import { UserService } from 'src/app/services/user.service';
 
 import { SignUpComponent } from './sign-up.component';
 
-describe('SignUpComponent', () => {
+describe('SignUpComponent should', () => {
   let component: SignUpComponent;
   let fixture: ComponentFixture<SignUpComponent>;
+  let userService: UserService;
+  let router: Router;
 
   beforeEach(async () => {
     await TestBed.configureTestingModule({
-      declarations: [ SignUpComponent ]
+      declarations: [SignUpComponent],
+      imports: [HttpClientTestingModule, ReactiveFormsModule, FormModule,
+        RouterTestingModule.withRoutes([
+          { path: 'sign-up', component: SignUpComponent },
+          { path: 'login', component: SignUpComponent }
+        ])],
+      providers: [UserService]
     })
-    .compileComponents();
+      .compileComponents();
   });
 
   beforeEach(() => {
     fixture = TestBed.createComponent(SignUpComponent);
     component = fixture.componentInstance;
+    userService = TestBed.get(UserService);
+    router = TestBed.get(Router);
     fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('not record when fields are empty', fakeAsync(() => {
+    const onClickMock = spyOn(component, 'signUp').and.callThrough();
+    component.form.setValue(SignUpMockEmpty);
+    fixture.detectChanges();
+    const button = fixture.debugElement.nativeElement.querySelector('button');
```

Comment:
You should select values by a id for more specificity. If you add another button it may failed